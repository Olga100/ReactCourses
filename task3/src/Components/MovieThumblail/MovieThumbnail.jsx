import React, {Component} from 'react';
import './MovieThumbnail.scss';

class MovieThumbnail extends Component {
    render() {
        const movie = this.props.movie;
        return (
            <div className="movieThumbnailContainer">
                <div className="poster">{this.props.movie.poster_path}</div>
                <div className="title-year">
                    <span> {this.props.movie.title}</span>
                    <span>{this.props.movie.release_date}</span>
                </div>
                <div>{this.props.movie.genres}</div>

            </div>
        )
    }
}

export default MovieThumbnail;


/*{
    "id": 0,
    "title": "string",
    "tagline": "string",
    "vote_average": 0,
    "vote_count": 0,
    "release_date": "string",
    "poster_path": "string",
    "overview": "string",
    "budget": 0,
    "revenue": 0,
    "runtime": 0,
    "genres": [
    "string"
]
}*/
