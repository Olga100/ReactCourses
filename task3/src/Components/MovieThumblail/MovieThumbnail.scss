.movieThumbnailContainer {
  display: flex;
  flex-direction: column;
  width: 240px;
  height: 420px;
  margin: 20px;

  .poster img{
    width: 240px;
    height: 330px;
  }

  .title-year {
    display: flex;
    justify-content: space-between;

    span {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 28px;
      width: 48px;
      border: 1px solid darkgray;
    }
  }
.genres {
  margin-right: 5px;
  font-size: 14px;
}

}