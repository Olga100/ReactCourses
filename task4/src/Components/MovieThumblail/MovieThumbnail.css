.movieThumbnailContainer {
  display: flex;
  flex-direction: column;
  width: 240px;
  height: 420px;
  margin: 20px; }
  .movieThumbnailContainer .poster img {
    width: 240px;
    height: 330px; }
  .movieThumbnailContainer .title-year {
    display: flex;
    justify-content: space-between; }
    .movieThumbnailContainer .title-year span {
      display: flex;
      justify-content: center;
      align-items: center;
      height: 28px;
      width: 48px;
      border: 1px solid darkgray; }
  .movieThumbnailContainer .genres {
    margin-right: 5px;
    font-size: 14px; }

/*# sourceMappingURL=MovieThumbnail.css.map */
