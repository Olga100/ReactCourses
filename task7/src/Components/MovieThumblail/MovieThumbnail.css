.movieThumbnailContainer {
  display: flex;
  flex-direction: column;
  width: 240px;
  height: 420px;
  margin: 20px; }
  .movieThumbnailContainer .link-wrapper {
    text-decoration: none;
    color: inherit; }
    .movieThumbnailContainer .link-wrapper .poster img {
      width: 240px;
      height: 330px; }
    .movieThumbnailContainer .link-wrapper .title-year {
      display: flex;
      justify-content: space-between; }
      .movieThumbnailContainer .link-wrapper .title-year span {
        display: flex;
        justify-content: center;
        align-items: center;
        height: 28px;
        width: 48px;
        border: 1px solid darkgray; }
    .movieThumbnailContainer .link-wrapper .genres {
      margin-right: 5px;
      font-size: 14px; }

/*# sourceMappingURL=MovieThumbnail.css.map */
