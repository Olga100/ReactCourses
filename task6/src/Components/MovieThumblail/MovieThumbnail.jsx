import React, {Component} from 'react';
import PropTypes from 'prop-types';

import './MovieThumbnail.scss';

class MovieThumbnail extends Component {

    getYear(isoDate) {
        return new Date(isoDate).getFullYear();
    }

    renderGenres(arr) {
        return arr.map((item) => (<span className="genres" key={item}>{item}</span>));
    }

    render() {
        const {poster_path, title, release_date, genres} = this.props.movie;
        const onMovieSelected = this.props.onMovieSelected;

        return (
            <div className="movieThumbnailContainer" onClick={() => onMovieSelected(this.props.movie)}>
                <div className="poster"><img src={poster_path} alt="poster"/></div>
                <div className="title-year">
                    <div> {title}</div>
                    <span>{this.getYear(release_date)}</span>
                </div>
                <div>{this.renderGenres(genres)}</div>

            </div>
        )
    }
}

MovieThumbnail.propTypes = {
    movie: PropTypes.object.isRequired,
    onMovieSelected: PropTypes.func
};

export default MovieThumbnail;
