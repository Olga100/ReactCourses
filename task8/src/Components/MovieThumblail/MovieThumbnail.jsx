// @flow
import React, {Component} from 'react';
import type {Node} from 'react';
import {Link} from 'react-router-dom';

import './MovieThumbnail.scss';

type Props = {
    movie: {
        id: number,
        title: string,
        poster_path: string,
        release_date: string,
        genres: Array<string>
    }
};

class MovieThumbnail extends Component<Props> {

    getYear(isoDate: string) {
        return new Date(isoDate).getFullYear();
    }

    renderGenres(arr: Array<string>): Node {
        return arr.map<Node>((item) => (<span className="genres" key={item}>{item}</span>));
    }

    render() {
        const {poster_path, title, release_date, genres} = this.props.movie;

        return (
            <div className="movieThumbnailContainer">
                <Link to={"/film/" + this.props.movie.id} className="link-wrapper">
                    <div className="poster"><img src={poster_path} alt="poster"/></div>
                    <div className="title-year">
                        <div> {title}</div>
                        <span>{this.getYear(release_date)}</span>
                    </div>
                    <div>{this.renderGenres(genres)}</div>
                </Link>
            </div>

        )
    }
}

export default MovieThumbnail;
