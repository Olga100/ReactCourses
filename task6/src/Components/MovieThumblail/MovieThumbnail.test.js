import React from 'react';
import renderer from 'react-test-renderer';

import MovieThumbnail from './MovieThumbnail';


describe ('MovieThumbnail', () => {

    const movie = {
        title: "title",
        poster_path: "http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg",
        genres: ["horror"],
        release_date: "2011-10-05T14:48:00.000Z"
    };

    it('should match snapshot with data', () => {
        const component = renderer.create(
            <MovieThumbnail  movie={movie}/>
        );
        const json = component.toJSON();

        expect(json).toMatchSnapshot();
    });

});