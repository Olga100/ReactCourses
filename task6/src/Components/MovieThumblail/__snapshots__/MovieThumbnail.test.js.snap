// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieThumbnail should match snapshot with data 1`] = `
<div
  className="movieThumbnailContainer"
>
  <div
    className="poster"
  >
    <img
      alt="poster"
      src="http://www.theamazingpics.com/wp-content/uploads/2014/05/Amazing-Picture-of-A-Japanese-Garden-in-Portland-USA.jpg"
    />
  </div>
  <div
    className="title-year"
  >
    <div>
       
      title
    </div>
    <span>
      2011
    </span>
  </div>
  <div>
    <span
      className="genres"
    >
      horror
    </span>
  </div>
</div>
`;
